tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts

```diff
@@ -203,6 +203,25 @@ export function hasModifierTag(apiItem: ApiItem, tagName: string): boolean {
 	return getModifierTags(apiItem).has(tagName);
 }
 
+/**
+ * Checks if the provided API item or any ancestors is tagged with the specified
+ * {@link https://tsdoc.org/pages/spec/tag_kinds/#modifier-tags | modifier tag}.
+ *
+ * @param apiItem - The API item whose documentation is being queried.
+ * @param tagName - The TSDoc tag name being queried for.
+ * Must be a valid TSDoc tag (including starting with `@`).
+ *
+ * @throws If the provided TSDoc tag name is invalid.
+ *
+ * @public
+ */
+export function ancestryHasModifierTag(apiItem: ApiItem, tagName: string): boolean {
```

Comment:
I was thinking something like this was probably needed. Thanks for adding. Not for this PR, but we should probably utilize this for other modifier notices as well (deprecated and legacy at least).